### DISCUSSION TITLE
How to define an openapi route that retuns an stream? (getting a type error with the latest versions)

### DISCUSSION THREAD

**Discussion by cldsnchz** [honojs/hono#1803] [Q&A] [answered]
Hi, with the latest versions (`hono` 3.11.4 and `@hono/zod-openapi` 0.9.3) when I define a openapi route that returns a stream, I'm getting the error: `Type 'Promise<Response>' is not assignable to type 'HandleTypeResponse<response>'`

for example:
```
app.openapi(route, (c) => {
  return c.streamText(async (stream) => {
    await stream.writeln('Hello') // Write a text with a new line ('\n')
    await stream.sleep(1000) // Wait 1 second
    await stream.write(`Hono!`) // Write a text without a new line
  })
})
```

Note that in the route I'm defining `responses: {}` because not sure how the stream has to be configure, but that was working in previous versions.

**Comment by cldsnchz** [comment]
I "fixed" the type by defining a response without content:
```
responses: { 
  200: {
    description: "...",
  },
},
```

Is that the expected way?. As of now is there a way to define the content of the stream? For example in my case, I return a list of events in an specific json format, so it would be nice if I can define that format, maybe not useful for validation but it's good for documentation.

**Comment by yusukebe** [comment, marked as answer]
Hi @cldsnchz 

> Is that the expected way?

Yes, this is expected. If you define types, it requires returning JSON in current implementation.
